platform/reworkd_platform/web/api/agent/tools/sidsearch.py

```diff
@@ -122,9 +122,31 @@ async def call(
             ]
         except Exception as e:
             logger.exception(e)
-            return stream_string("Unable to fetch SID results", True)
+            return None
 
         if not snippets:
-            return stream_string("No good results found by SID", True)
+            return None
 
         return summarize_sid(self.model, self.language, goal, task, snippets)
+
+
+    async def call(
+        self,
+        goal: str,
+        task: str,
+        input_str: str,
+        user: UserBase,
+        oauth_crud: OAuthCrud,
+        *args: Any,
+        **kwargs: Any,
+    ) -> FastAPIStreamingResponse:
+        res = await self._run_sid(
+            goal, task, input_str, user, oauth_crud
+        )
+        if res is None:
+            # fall back to search if no results are found
+            return await Search(self.model, self.language).call(
+                goal, task, input_str, user, oauth_crud
+            )
+        else:
+            return res
```

Comment:
Lets get ride of the else statement here. 
```suggestion
         # fall back to search if no results are found
        return await self._run_sid(goal, task, input_str, user, oauth_crud) or await Search(self.model, self.language).call(
        goal, task, input_str, user, oauth_crud
    )
```